bcl2fastq/lib/bcl2fastq_utils.py

```diff
@@ -297,6 +297,15 @@ def construct_command(self):
         """
         raise NotImplementedError("Subclasses should implement this!")
 
+    def delete_output(self):
+        """
+        Delete the output directory if it exists
+        :return: None
+        """
+        if os.path.isdir(self.config.output):
+            log.info("Found a directory at output path {}, will remove it.".format(self.config.output))
+            shutil.rmtree(self.config.output)
```

Comment:
I can implement this. It is probably the best alternative, though I'm sort of conflicted since it looks down the interface a bit more than previously (when it was made to quite closely match how the command line program works). That aside it is probably a good idea to do something like this, the solution proposed here was honestly mostly because I felt a bit pressed on time in getting something working for this.